product_customer_pricelist/models/sale.py

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2018 Daniel Campos <danielcampos@avanzosc.es> - Avanzosc S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def action_confirm(self):
+        res = super(SaleOrder, self).action_confirm()
+        if res:
+            for order in self:
+                for pricelist in order.mapped(
```

Comment:
porque haces un for, no sería suficiente con poner el primero o el ultimo ignorando aquellos que no tengan?